spark/core.py

```diff
@@ -310,32 +314,118 @@ def clear_rect(self, *args):
 
     # Draws circle at given coordinates
     def circle(self, *args):
-        self.check_coords("circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.fill_arc(*arc_args)
-        self.canvas.stroke_arc(*arc_args)
+        self.check_coords("circle", *args, width_only = True)
+        self.ellipse(*args, args[2])
 
     # Draws filled circle
     def fill_circle(self, *args):
-        self.check_coords("fill_circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.fill_arc(*arc_args)
+        self.check_coords("fill_circle", *args, width_only = True)
+        self.fill_ellipse(*args, args[2])
 
     # Draws circle stroke
     def stroke_circle(self, *args):
-        self.check_coords("stroke_circle", *args, width_only=True)
-        arc_args = self.arc_args(*args)
-        self.canvas.stroke_arc(*arc_args)
-        
+        self.check_coords("stroke_circle", *args, width_only = True)
+        self.fill_ellipse(*args, args[2])
+
+    def ellipse(self, *args):
+        self.check_coords("ellipse", *args)
```

Comment:
Any performance concerns over these redundant `check_coords` calls (between `circle` -> `ellipse` -> `fill_ellipse`)?
We could optimize it by adding a hidden parameter/flag that disables these validation checks, but it may throw monkey wrenches in our logic.

Reply:
As `check_coords` is only a check on the correct number and type, I don't think it'll have a super significant performance increase, but if it can be optimized simply that'd be great.